patroni/config.py

```diff
@@ -201,7 +202,10 @@ def _load_config_path(self, path: str) -> Dict[str, Any]:
         overall_config: Dict[str, Any] = {}
         for fname in files:
             with open(fname) as f:
-                config = yaml.safe_load(f)
+                config: dict[str, Any] = yaml.safe_load(f)
+                if not isinstance(config, dict):  # pyright: ignore [reportUnnecessaryIsInstance]
+                    logger.error('%s does not contain a dict', fname)
+                    raise ConfigParseError('invalid config file {0}'.format(fname))
                 patch_config(overall_config, config)
```

Comment:
```suggestion
```suggestion
                config = yaml.safe_load(f)
                if not isinstance(config, dict):
                    logger.error('%s does not contain a dict', fname)
                    raise ConfigParseError(f'invalid config file {fname}')
                patch_config(overall_config, cast(Dict[Any, Any], config))
```